### PR TITLE
[v7r0] Sleep and retry if the FC is unavailable when uploading output

### PR DESCRIPTION
**Pull request by chaen**
We have a design flaw which is that, when uploading the output of a job, if it fails, we just upload it somewhere else and create a failover request. However, in the case where the DFC is totally down, you can't even reach this state, because you cannot perform the `hasAccess` operation. 
In practice, this results in losing the result of your job. 
This PR introduces a sleep & retry in case this happens
@fstagni this is what we discussed

BEGINRELEASENOTES
*DMS
CHANGE: failoverTransfer sleep & retries in case of FC unavailability

ENDRELEASENOTES

### CODE REVIEW

File: DataManagementSystem/Client/FailoverTransfer.py [hunk 1]
```diff
@@ -72,18 +75,33 @@ def transferAndRegisterFile(self,
...
+                                                                guid=fileGUID,
+                                                                checksum=fileChecksum)
+        self.log.verbose(result)
+
+        # If the FC is unavailable, we stay in the loop and retry
+        # otherwise we continue without retrying
+        if result['OK'] or not cmpError(result, EFCERR):
+          break
+        self.log.error("transferAndRegisterFile: FC unavailable, retry")
+        time.sleep(sleeptime)
```
Comment (andresailer) [inline DataManagementSystem/Client/FailoverTransfer.py:103]: Is it necessary to disable the JobWrapper Watchdog CPU consumption monitoring here?

Comment (chaen) [reply DataManagementSystem/Client/FailoverTransfer.py:103]: I must admit being totally unknowledgeable here, but I was under the impression that the download/upload were always done with the Watchdog disabled ?

Comment (andresailer) [reply DataManagementSystem/Client/FailoverTransfer.py:103]: Indeed.
https://gitlab.cern.ch/lhcb-dirac/LHCbDIRAC/-/blob/master/LHCbDIRAC/Workflow/Modules/UploadOutputData.py#L232-233

https://gitlab.cern.ch/CLICdp/iLCDirac/ILCDIRAC/-/blob/Rel-v29r2/Workflow/Modules/UploadOutputData.py#L180-183

But not in DIRAC itself, as far as I can tell.

Comment (chaen) [reply DataManagementSystem/Client/FailoverTransfer.py:103]: ah ok. Well if it isn't disabled before, I don't see why i should be now. Also, this should be an extremely rare case !

Comment (andresailer) [reply DataManagementSystem/Client/FailoverTransfer.py:103]: Yes, agreed. The disabling needs to happen at a higher level than this function here.

**Pull request merged**